packages/cactus-plugin-ledger-connector-iroha/src/main/typescript/plugin-ledger-connector-iroha.ts

```diff
@@ -0,0 +1,684 @@
+              assetId: req.params[1],
+              pageSize: req.params[2],
+              firstTxHash: req.params[3],
+            },
+          );
+          return { transactionReceipt: response };
+        } catch (err) {
+          throw new Error(err);
+        }
+      }
+      case IrohaQuery.GetBlock: {
+        try {
+          const response = await queries.getBlock(queryOptions, {
+            height: req.params[0],
+          });
+          return { transactionReceipt: response };
+        } catch (err) {
+          throw new Error(err);
+        }
+      }
+      case IrohaCommand.CallEngine: {
+        try {
+          const response = await commands.callEngine(commandOptions, {
+            type: req.params[0],
+            caller: req.params[1],
+            callee: req.params[2],
+            input: req.params[3],
+          });
+          return { transactionReceipt: response };
+        } catch (err) {
+          throw new Error(err);
+        }
+      }
+      case IrohaQuery.GetEngineReceipts: {
+        try {
+          const response = await queries.getEngineReceipts(queryOptions, {
+            txHash: req.params[0],
+          });
+          return { transactionReceipt: response };
+        } catch (err) {
+          throw new Error(err);
+        }
+      }
+      case IrohaQuery.FetchCommits: {
+        try {
+          const response = await queries.fetchCommits(queryOptions);
+          return { transactionReceipt: response };
+        } catch (err) {
+          throw new Error(err);
+        }
+      }
+      case IrohaCommand.AddPeer: {
+        try {
+          const response = await commands.addPeer(commandOptions, {
+            address: req.params[0],
+            peerKey: req.params[1],
+          });
+          return { transactionReceipt: response };
+        } catch (err) {
+          throw new Error(err);
+        }
+      }
+      case IrohaCommand.RemovePeer: {
+        try {
+          const response = await commands.removePeer(commandOptions, {
+            publicKey: req.params[0],
+          });
+          return { transactionReceipt: response };
+        } catch (err) {
+          throw new Error(err);
+        }
+      }
+      case IrohaQuery.GetPeers: {
+        try {
+          const response = await queries.getPeers(queryOptions);
+          return { transactionReceipt: response };
+        } catch (err) {
+          throw new Error(err);
+        }
+      }
+      case "producePendingTx": {
+        try {
+          const response = await commands.createAsset(commandOptions2, {
+            assetName: "coinb",
+            domainId: "test",
+            precision: 3,
+          });
+          return { transactionReceipt: response };
+        } catch (err) {
+          throw new Error(err);
+        }
+      }
+      default: {
+        return {
+          transactionReceipt:
+            "command or query does not exist, or is not supported in current version",
```

Comment:
@hxlaf `throw new RuntimeError("the message")` instead of returning valid looking response object that might confuse the callers.

Reply:
Applied the throw via `throw new RuntimeError("the message")`. Thank you!
![image](https://user-images.githubusercontent.com/10954157/129980222-1658bc18-1c66-40b8-89c7-436962562393.png)